brkt_cli/__init__.py

```
@@ -197,6 +157,147 @@ def main():
             log.error('Interrupted by user')
     return 1
 
+
+def command_update_encrypted_ami(values, log):
+    encrypted_ami_name = None
+    if values.encrypted_ami_name:
+        try:
+            aws_service.validate_image_name(values.encrypted_ami_name)
+            encrypted_ami_name = values.encrypted_ami_name
+        except aws_service.ImageNameError as e:
+            print(e.message, file=sys.stderr)
+            return 1
+    region = values.region
+    if util.validate_region(region) == 1:
+        print ('Invalid region %s' % region,
+               file=sys.stderr)
+        return 1
+    nonce = util.make_nonce()
+    default_tags = encrypt_ami.get_default_tags(nonce, '')
+    aws_svc = aws_service.AWSService(
+        nonce, None, default_tags=default_tags)
+    aws_svc.connect(region)
+    zones = [str(z.name) for z in aws_svc.conn.get_all_zones()]
+    # If supplied, use the zone in the args
+    if values.zone:
+        zone = values.zone
+        if zone not in zones:
+            print (
+                'Invalid zone %s for region %s. ' % (zone, region) +
+                'Must be one of %s' % str(zones),
+                file=sys.stderr
+            )
+            return 1
+    else:
+        zone = zones[0]
+        log.info('Using zone %s', zone)
+    encrypted_ami = values.ami
+    guest_ami_error = aws_svc.validate_guest_ami(encrypted_ami)
+    if guest_ami_error:
+        print(guest_ami_error, file=sys.stderr)
+        return 1
+    updater_ami = values.updater_ami
+    updater_ami_error = aws_svc.validate_encryptor_ami(values.updater_ami)
```

Users can make mistakes, and we need to fail early and clearly when they do.  I'm pretty sure that we tag the encrypted AMI.  Can you check the tags during validation?